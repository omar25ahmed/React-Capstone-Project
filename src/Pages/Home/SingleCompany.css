.company {
  list-style: none;
  cursor: pointer;
  height: 150px;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  transition: all 0.4s;
}

li {
  background-color: #4066ae;
}

li:nth-child(3n - 2) {
  background-color: #4063a5;
}

.symbol {
  color: #2d4472;
  font-size: 50px;
  font-style: italic;
}

.title {
  font-weight: bold;
  font-size: 10px;
}

.company:hover {
  background-color: #164fb9;
}
