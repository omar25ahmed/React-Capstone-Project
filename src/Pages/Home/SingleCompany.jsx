import React from 'react';
import { useNavigate } from 'react-router-dom';
import PropTypes from 'prop-types';
import './SingleCompany.css';
import { FaArrowAltCircleRight } from 'react-icons/fa';

function SingleCompany({ company }) {
  const { companyName, symbol } = company;
  const navigate = useNavigate();
  const openCompanyProfileHandler = () => {
    navigate(`/${companyName}`);
  };
  return (
    <li data-testid="company" aria-hidden="true" onClick={openCompanyProfileHandler} className="col-6 col-md-3 company">
      <div className="symbol">{symbol}</div>
      <div className="title">{companyName}</div>
      <div className="icon"><FaArrowAltCircleRight /></div>
    </li>
  );
}

export default SingleCompany;

SingleCompany.defaultProps = {
  company: {
    companyName: '',
    symbol: '',
  },
};

SingleCompany.propTypes = {
  company: PropTypes.shape({
    companyName: PropTypes.string,
    symbol: PropTypes.string,
  }),
};
