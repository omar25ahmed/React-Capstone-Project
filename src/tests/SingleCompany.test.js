import renderer from 'react-test-renderer';
import { render, screen } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import store from '../Redux/Store';
import SingleCompany from '../Pages/Home/SingleCompany';

describe('testing Home component', () => {
  it('snapshot for the Single Company Component ', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <Router>
            <SingleCompany />
          </Router>
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('testing Single Company component renders correctly', () => {
    render(
      <Provider store={store}>
        <Router>
          <SingleCompany />
        </Router>
      </Provider>,
    );
    expect(screen.getByTestId('company')).toBeInTheDocument();
  });
  it('should GET data', async () => {
    render(
      <Provider store={store}>
        <Router>
          <SingleCompany />
        </Router>
      </Provider>,
    );

    const listItemElements = await screen.findAllByTestId('company');
    expect(listItemElements).not.toHaveLength(0);
  });
});
