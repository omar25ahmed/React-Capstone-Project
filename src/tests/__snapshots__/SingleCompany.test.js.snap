// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Home component snapshot for the Single Company Component  1`] = `
<li
  aria-hidden="true"
  className="col-6 col-md-3 company"
  data-testid="company"
  onClick={[Function]}
>
  <div
    className="symbol"
  />
  <div
    className="title"
  />
  <div
    className="icon"
  >
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M256 8c137 0 248 111 248 248S393 504 256 504 8 393 8 256 119 8 256 8zM140 300h116v70.9c0 10.7 13 16.1 20.5 8.5l114.3-114.9c4.7-4.7 4.7-12.2 0-16.9l-114.3-115c-7.6-7.6-20.5-2.2-20.5 8.5V212H140c-6.6 0-12 5.4-12 12v64c0 6.6 5.4 12 12 12z"
      />
    </svg>
  </div>
</li>
`;
